### PR TITLE
Fix return validation

### LINKED ISSUE #267: Valid docs rule does not support closure parameters
The rule that validates documentation blocks does not support closure type parameters and mistakes them with return values. 
I think it is probably related to this func:
```
func delcarationReturns(declaration: String, kind: SwiftDeclarationKind) -> Bool {
    return declaration.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
}
```
The `->` that exists in the closure parameter is mistaken by the return `->`.

### PR DESCRIPTION
**Pull request by diogoguimaraes**
Closes #267.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/ValidDocsRule.swift [hunk 1]
```diff
@@ -38,7 +38,11 @@ func superfluousOrMissingThrowsDocumentation(declaration: String, comment: Strin
 }
 
 func delcarationReturns(declaration: String, kind: SwiftDeclarationKind) -> Bool {
-    return declaration.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
+    let outsideBraces = NSMutableString(string: declaration)
+    regex("(\\s*->\\s*).*[^(]\\)").replaceMatchesInString(outsideBraces, options: [],
+        range: NSRange(location: 0, length: outsideBraces.length), withTemplate: "")
+
+    return outsideBraces.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/ValidDocsRule.swift]: Could you return early if `SwiftDeclarationKind.variableKinds().contains(kind)`? Much cheaper than always running the regex replace.

**Comment by jpsim** [comment]
Adding some test cases to `triggeringExamples` and `nonTriggeringExamples` would be appreciated. Thanks!

**Comment by diogoguimaraes** [comment]
Updated @jpsim.

**Comment by jpsim** [comment]
Thanks! I've invited you to be a contributor to SwiftLint to make it easier for you to continue contributing, I hope that's ok.

**Comment by diogoguimaraes** [comment]
Thank you for inviting me! It will be great to contribute :)  
Can you please just create a minor release with this fix?

**Comment by jpsim** [comment]
> Can you please just create a minor release with this fix?

Sure, I just made one for SourceKitten so now I'm testing that with SwiftLint (#278), and if that's all green, I'll cut SwiftLint 0.5.2 right after.

**Comment by jpsim** [comment]
[SwiftLint 0.5.2: Snuggle™](https://github.com/realm/SwiftLint/releases/tag/0.5.2) is out! Homebrew PR pending: https://github.com/Homebrew/homebrew/pull/47156

**Pull request merged**